clients/src/main/java/org/apache/kafka/common/requests/CreateTopicsResponse.java

```diff
@@ -84,63 +44,43 @@
      * POLICY_VIOLATION(44)
      */
 
-    private final Map<String, ApiError> errors;
-    private final int throttleTimeMs;
+    private final CreateTopicsResponseData data;
 
-    public CreateTopicsResponse(Map<String, ApiError> errors) {
-        this(DEFAULT_THROTTLE_TIME, errors);
+    public CreateTopicsResponse(CreateTopicsResponseData data) {
+        this.data = data;
     }
 
-    public CreateTopicsResponse(int throttleTimeMs, Map<String, ApiError> errors) {
-        this.throttleTimeMs = throttleTimeMs;
-        this.errors = errors;
+    public CreateTopicsResponse(Struct struct, short version) {
+        this.data = new CreateTopicsResponseData(struct, version);
```

Comment:
We moved away at some point from passing the version in the constructor. The idea was do the same for the request classes so that we could remove the builders altogether. What's the reasoning for reintroducing this?

Reply:
We need the version in order to parse the data that is in the `Struct` object.  Without the version, we don't know the format that we should expect.  Similarly, once this is converted to using ByteBuffers, we'll need the version before we can parse what is in the ByteBuffer.

The version is not stored in the response.  Responses should continue to represent the data, not a particular serialization of the data.  It would be nice if Request was the same way-- at some point we'll probably fix those classes to be like that too.

Reply:
I see. This intermediate step is a bit weird since we use the `version` to create the `Struct` in the first place. Once we drop `Struct`, then it seems reasonable to have a `parse` method that takes a `ByteBuffer` and `version` and returns a `CreateTopicsResponse`.